### ISSUE TITLE
Three phase version

### ISSUE THREAD

**Issue by aleonnet** [Savjee/home-energy-monitor#23]
Hi @Savjee great job!
It would be very good if you co

uld include a three phase version. I mean read the current and voltage for the three phases. These would open the possibility for several other applications and house sizes!

Does the EmonLib-esp32 support the three phase calculations?

By the way I will open source my small board to adapt current from the sct-013-000.
![sct-013-000](https://user-images.githubusercontent.com/20618327/80531459-14b5f400-8971-11ea-820f-014f58c40f7a.png)

**Comment by Savjee** [comment]
A 3-phase system requires 3 separate CT sensors. This should be easy to achieve because the ESP32 has enough ADC inputs. The only thing required is enough input pins for the sensors ;)

The library doesn't need to be changed. You can create 3 instances of it and have them work together.

Cool! What software are you using to design that PCB? I started mine on EasyEDA but didn't finish it yet. Hard to find components like headphones jacks with correct footprints for EasyEDA. How are you tackling this?

**Comment by aleonnet** [comment]
I'm afraid the power factor for each phase would not be calculated correctly, by simply measuring the current without its corresponding voltage. Maybe this can be adjusted by software. What do you think?
I'm using EasyEDA as well and I its really not easy :) to find some good components' footprints. However you can use a pachymeter to check your actual components. For example, I had to adjust the hole size of the stereo jack with it, note that it's larger.

**Comment by Savjee** [comment]
Indeed, this is my current problem. I need to add support for a voltage sensor as well. Then the setup would work:

- 3 CT clamps for measuring current
- 1 voltage sensor

Now you can:
- Measure voltage
- Measure current with 3 emonlib instances (while using the actual voltage measurement)

That should give fairly good results. From what I read on OpenEnergyMonitor, it requires an AC-AC power adapter: https://learn.openenergymonitor.org/electricity-monitoring/voltage-sensing/measuring-voltage-with-an-acac-power-adapter

Glad I'm not the only one struggling with EasyEDA. Problem is that I don't have all my components yet. Still waiting on a few from AliEpress...

**Comment by gieljnssns** [comment]
@Savjee Any updates on this?

**Comment by Savjee** [comment]
I've abandoned the idea of creating a custom PCB. Maybe @aleonnet has finished his?

**Comment by gieljnssns** [comment]
@Savjee 
I mean the three phase version…
And the support for the voltage sensor.

**Comment by Savjee** [comment]
No updates on those fronts, unfortunately.

**Comment by gieljnssns** [comment]
@Savjee 
> A 3-phase system requires 3 separate CT sensors. This should be easy to achieve because the ESP32 has enough ADC inputs. The only thing required is enough input pins for the sensors ;)

What do you mean with this?
Which pins can be used for ADC? What numbering do I have to use?
https://esp32.com/viewtopic.php?t=13145
![image](https://user-images.githubusercontent.com/17234951/102540818-e7f54f00-40af-11eb-9416-10a8a74816a6.jpeg)

**Comment by JamesLeonDufour** [comment]
I'll be interested to follow up on that one. Here is a good article . http://www.technik-fan.de/index.php/Open_Energy_Monitor_mit_dem_ESP32
That would require an esp 32 not 8266 right ?

**Comment by Savjee** [comment]
Yes, the ESP8266 only has 1 ADC channel and you would need 3.
There are two options:
1. Switch to an ESP32 which has 18 channels in total (8 + 10)
2. Use the ESP8266 with an external ADC.

Regardless of which option you choose, an external ADC will always give you better accuracy compared to the built-in ADC.

**Comment by JamesLeonDufour** [comment]
Hi Savjee, 

Thanks for taking the time to answer, I have been digging quiet a lot on CT clamp and did my first iteration.

![WhatsApp Image 2022-10-11 at 9 57 40 PM](https://user-images.githubusercontent.com/6214012/195187295-58221f12-15dc-4597-b887-f2c2b8ae9dcc.jpeg)

This is based on your project. Luckily I ordered a SCT-013-030 30a 1v that I assume you are using in your project and if I understand correctly after my reserach doesnt need a burden resistor and seem to make the job a lot easier for everyone (instead of output a ma) ?

So i connected the whole thing, added an Oled screen as you can see and was going to work on the IOT platform, MQTT etc later to transmit data.

Just when I wanted to install it, I realised that I had 3 phases in my house (Belguim), so I continued my digging. 

Using the ESP32 option(I have one), the easy way I found is basically replicate the same approach like this example and yours x3. Pretty straight forward. http://www.technik-fan.de/index.php/Open_Energy_Monitor_mit_dem_ESP32.

Other option you are mentionning would be connecting an ADS1115 ? to 3 SCT clamp. But the 0-1V model like the one i have or should i switch back to a 000 series (in ma).

You are mentionning the external ADC (eg: ads1115) would be more accurate, but would I still be able to read each phases independently ? (Could be interesting to monitor which part of the house and plugs consume more).

There is this nice pieces of circuitery as well 4 channel, https://www.mottramlabs.com/esp_products.html but seems a bit of lot of complication with the 2 options listed before.

Lets me know which approach you would follow first ? the easiest ?  ( i assume the ESP8266 + ADS1115 + SCT013-030)

Thanks in advance, this is a very debatted project and seems to be a lot of confusion around the SCT CT clamp different model and the whole interfacing and not really clear project for 3 phases at least.

Best

**Comment by Savjee** [comment]
In any case, I would advice to use the SCT013-030 (with 0-1V output). The voltage output can easily be measured by any ADC.

I believe the easiest solution would be to use an ESP32 with the built-in ADC. This requires no additional hardware, aside from the CT sensors. ADC1 has 8 channels (useable while WiFi is active), which mean you can attach up to 8 CT sensors. You can use ADC2 as well, but then you cannot use WiFi (interesting when you use ethernet).

Using an external ADC allows you to use both an ESP32 or ESP8266. However, the mentioned chip (ADS1115) only has 4 channels. That's enough for 3-phase, but not enough if you want to monitor individual circuits.
